server/gitrest/packages/gitrest-base/src/routes/summaries.ts

```diff
@@ -148,44 +148,46 @@ async function createSummary(
 
 	// Waiting to pre-compute and persist latest summary would slow down document creation,
 	// so skip this step if it is a new document.
-	if (!isNew && isContainerSummary(payload)) {
-		const latestFullSummary: IWholeFlatSummary | undefined = await wholeSummaryManager
-			.readSummary(writeSummaryResponse.id)
-			.catch((error) => {
-				// This read is for Historian caching purposes, so it should be ignored on failure.
-				Lumberjack.error(
-					"Failed to read latest summary after writing container summary",
-					lumberjackProperties,
-					error,
-				);
-				return undefined;
-			});
-		if (latestFullSummary) {
-			if (persistLatestFullSummary) {
-				try {
-					// TODO: does this fail if file is open and still being written to from a previous request?
-					await persistLatestFullSummaryInStorage(
-						fileSystemManager,
-						getFullSummaryDirectory(
-							repoManager,
-							repoManagerParams.storageRoutingId.documentId,
-						),
-						latestFullSummary,
+	if (isContainerSummary(payload)) {
+		const latestFullSummary: IWholeFlatSummary | undefined = (
+			writeSummaryResponse as IWholeFlatSummary
+		).trees
+			? writeSummaryResponse
+			: await wholeSummaryManager.readSummary(writeSummaryResponse.id).catch((error) => {
+					// This read is for Historian caching purposes, so it should be ignored on failure.
+					Lumberjack.error(
+						"Failed to read latest summary after writing container summary",
 						lumberjackProperties,
+						error,
 					);
-				} catch (error) {
+					return undefined;
+			  });
+		if (latestFullSummary) {
+			if (persistLatestFullSummary) {
+				// Send latest full summary to storage for faster read access.
+				const persistP = persistLatestFullSummaryInStorage(
+					fileSystemManager,
+					getFullSummaryDirectory(
+						repoManager,
+						repoManagerParams.storageRoutingId.documentId,
+					),
+					latestFullSummary,
+					lumberjackProperties,
+				).catch((error) => {
+					// Persisting latest summary is an optimization, not a requirement, so do not throw on failure.
 					Lumberjack.error(
 						"Failed to persist latest full summary to storage during createSummary",
 						lumberjackProperties,
 						error,
 					);
-					// TODO: Find and add more information about this failure so that Scribe can retry as necessary.
-					throw new NetworkError(
-						500,
-						"Failed to persist latest full summary to storage during createSummary",
-					);
+				});
+				if (!isNew) {
```

Comment:
I have a similar question.
Because if Historian cache expires, and multiple clients connect to a new document at the same time, connectDoc will again take long (because getSummary() would take long causing joinOp errors), right?

Reply:
> Because if Historian cache expires, and multiple clients connect to a new document at the same time, connectDoc will again take long (because getSummary() would take long causing joinOp errors), right?

@sharmakhushboo  This is irrelevant to this PR. Historian cache expiring will always mean that the next get req will hit storage. That's why we store the pre-computed summary so that the next read is faster.

Reply:
Thanks @znewton! I might be missing something very obvious here, so I appreciate the patience with my questions.

> We always persist if there is a latestFullSummary to persist.

I see that we would actually execute and await for `persistP` in case this is not an initial summary upload (if condition I asked about). But per your comment, I understand that we would always execute that promise - we just might not await for it in the other condition, making it work like fire and forget. Where is that happening? In other words, the part I'm confused about: we only execute the promise (which happens to be with `await`) within the if condition, and I am missing the part where we do it for the other scenario.

Reply:
Hey @hedasilv, no worries. I think what's maybe confusing here is that `persistP` is simply tracking the `Promise` that is returned by executing `persistLatestFullSummaryInStorage()` on line 168. `persistLatestFullSummaryInStorage` is _always_ executed if `latestFullSummary && persistLatestFullSummary`.

The only part that is different for `isNew` is that we `await` the promise when `!isNew`, meaning that we pause further processing of the request until persisting ends. When `isNew === true`, we instead _don't_ wait for the `Promise` returned  by `persistLatestFullSummaryInStorage` to resolve/reject, and instead immediately proceed to return the `latestFullSummary` to Historian for caching, thus allowing the CreateDocument flow to happen with no persisting-related slowdown.

Reply:
Thanks @znewton! Indeed, I share the same understanding. I think what confused me is: I missed the fact that calling 

```ts
const persistP = persistLatestFullSummaryInStorage(...).catch(...);
```

would already work as a "fire and forget" strategy for persistLatestFullSummaryInStorage(). In other words, we don't need to do something like `void (persistP)()` to actually trigger the function. Thanks for the clarification!